projects/igniteui-angular/src/lib/grids/headers/grid-header.component.ts

```diff
@@ -189,6 +184,7 @@ export class IgxGridHeaderComponent implements DoCheck, OnInit, OnDestroy {
 
     public onFilteringIconClick(event) {
         event.stopPropagation();
+        this.grid.filteringService.initFilteringSettings();
```

Comment:
`initFilteringSettings` should be part of `toggleFilterDropdown`, since if you toggle the menu with keyboard the `initFilteringSettings` won't be executed.

Reply:
Moved